aten/src/ATen/core/IListRef.h

```diff
@@ -0,0 +1,612 @@
+#pragma once
+
+#include <ATen/core/ivalue_to.h>
+#include <c10/util/Exception.h>
+
+#include <functional>
+#include <initializer_list>
+#include <iterator>
+#include <type_traits>
+
+/*
+ * Adding support for a new Tag
```

Comment:
nit: this comment is kind of in the wrong place, it'd be better to see the high level comment for what an IListRef is first haha